### ISSUE TITLE
Some issues using latest docker image.

### ISSUE THREAD

**Issue by ruseinov** [spark-jobserver/spark-jobserver#493]
First of all I had to modify the config in order for this thing to stop throwing time-outs and errors at me:
```
spray.can.server {
  parsing.max-content-length = 50m
  idle-timeout = 400s
  request-timeout = 300s
}
deploy {
  manager-start-cmd = "./app/manager_start.sh"
}
```

But then, when I POST a jar to a jobserver I get:
```
Unexpected 0
```
altough jar appears in the list

Sometimes when I try to list jars I get ask timeout

When I try to run the job in a jar I get a Future timeout 60 seconds error.

I did investigate how it looks process-wise - there's a `manager_start.sh` being executed, but it exits really fast and then I get a Future timeout.

The job implements SparkJob trait and compiles just fine. Versions of spark and job-server libraries should also be all good. 

There is no log output whatsoever about what's going wrong with upload or manager_start.sh

**Comment by ruseinov** [comment]
Imagename: `velvia/spark-jobserver:0.6.2.mesos-0.28.1.spark-1.6.1`

**Comment by noorul** [comment]
I am also facing similar problem with docker image based on 0.6.1. I am trying to upload jar of size 100 mb with settings

`parsing.max-content-length = 200m`

But the upload request never finishes.

**Comment by velvia** [comment]
Are you able to split the jar up into application portion and dependencies?  Can the dependency jar be linked using —jars or dependent-jar-uris?

**Comment by noorul** [comment]
Doesn't this require spark-jobserver restart?

**Comment by velvia** [comment]
No….  dependent-jar-uris is a property of the CONTEXT config.   You can create a new context with updated dependent-jar-uris, no SJS restart is needed.   Only if you want dependent-jar-uris as part of the global SJS config do you need to restart SJS — but I think this is a bad idea anyways, it’s always better to localize dependencies on each context, rather than pollute all your contexts with it.

**Comment by noorul** [comment]
@velvia I think some jars (like cassandra connector) are supposed to be on driver classpath too. How do we solve that? Does dependent-jar-uris jar get loaded to driver JVM also?

**Comment by velvia** [comment]
Yes, dependent-jar-uris should, it uses the context’s addJars method.
I sort of remember fixing some issue in Spark where the context itself does not have the added jars.

**Comment by noorul** [comment]
I set 

`short-timeout = 60 s`

and then I could upload

**Comment by ruseinov** [comment]
Still, does not answer my question here, "Unexpected 0" on upload and jar present in the list, although not able to run it for some reason.

**Comment by noorul** [comment]
@ruseinov I tried to recreate "Unexpected O" message but in vain. Can you consistently re produce this? If so do you mind sharing the recipe?

**Comment by ruseinov** [comment]
Well, I got it using Postman to submit a file.

Pretty much submitted a 50 meg jar and got this in return.

**Comment by ruseinov** [comment]
And yes, it was a consistent error using a docker image.
